SmartDeviceLink/SDLURLRequestTask.m

```diff
@@ -55,7 +54,8 @@ + (instancetype)taskWithURLRequest:(NSURLRequest *)request completionHandler:(SD
 }
 
 - (void)dealloc {
-    [_connection cancel];
+    self.state = SDLURLRequestTaskStateCanceled;
```

Comment:
The canceled state seems to only be set for the split second before this object is dealloced. Is that really necessary?